tests/runner.py

```diff
@@ -663,9 +664,13 @@ def do_run(self, src, expected_output, args=[], output_nicerizer=None,
     for engine in js_engines:
       #print 'test in', engine
       js_output = self.run_generated_code(engine, filename + '.o.js', args, output_nicerizer=output_nicerizer, assert_returncode=assert_returncode)
+      js_output = js_output.replace('\r\n', '\n')
       try:
-        self.assertContained(expected_output, js_output.replace('\r\n', '\n'))
-        self.assertNotContained('ERROR', js_output)
+        if assert_identical:
+          self.assertIdentical(expected_output, js_output)
+        else:
+          self.assertContained(expected_output, js_output)
+          self.assertNotContained('ERROR', js_output)
```

Comment:
Hmm, how come updating the test required changing the harness?

Reply:
We can assume `stringToUTF8` to never output partial characters at all, but always properly truncate to full characters. That should be the case at https://github.com/kripken/emscripten/blob/master/src/preamble.js#L623, so we do not need to complicate for testing against a broken `stringToUTF8` here. Or are you saying you found a problem with `stringToUTF8` and had to work around that with the changes here?

Reply:
No, there were no problems with stringToUTF8.

OK, I'll remove the test for a bad truncation. Should I remove the "assert_identical" mode too?
My opinion - it's still useful, since it allows the test implementer to better express their intention.

Reply:
I think the assert_identical mode looks useful to have, but let's wait for @kripken for his thoughts as well.

Reply:
Yeah, seems useful to me too.